tool/src/org/antlr/v4/codegen/target/CppTarget.java

```diff
@@ -37,34 +36,24 @@
 		"switch", "template", "this", "thread_local", "throw", "true",
 		"try", "typedef", "typeid", "typename", "union", "unsigned",
 		"using", "virtual", "void", "volatile", "wchar_t", "while",
-		"xor", "xor_eq"
-	};
+		"xor", "xor_eq",
 
-	/** Avoid grammar symbols in this set to prevent conflicts in gen'd code. */
-	protected final Set<String> badWords = new HashSet<String>();
+		"rule", "parserRule"
```

Comment:
wow `parserRule` is a reserved word in C++?

Reply:
I don't think so, this word is presented in all targets and I don't know why. I'm trying to figure it out right now.

Reply:
hahaha, yeah, so weird